### PR TITLE
handle icons for status

### PR DESCRIPTION
**Pull request by FredLL-Avaiga**
## What type of PR is this? (check all applicable)

- [x] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description
Simplify the work done in the hacktoberfest PR

only new property is icons: Union[bool, list[str]]

if false, no icons,
if true: pre-defined icons
if list, for each element:
  - true: predefined icon
  - false: no icon
  - string: try to use as SVG URL (ends with `.svg`) or SVG string (starts with `<svg `)

## Related Tickets & Documents
- Related Issue #1157 

## How to reproduce the issue
```py
from taipy.gui import Gui

status = [
    ("warning", "Task is launched."),
    ("warning", "Taks is waiting."),
    ("error", "Task timeout."),
    ("success", "Task Succeded"),
    ("info", "Process was cancelled.")
]

page = """
<|{status}|status|>
<|{status}|status|icons|>
<|{status}|status|icons=https://www.svgrepo.com/show/530594/pants.svg;https://www.svgrepo.com/show/530595/hotel.svg;https://www.svgrepo.com/show/530596/diving-goggles.svg;https://www.svgrepo.com/show/530597/hat.svg|>
<|{status}|status|icons=https://www.svgrepo.com/show/530594/pants.svg;false;true;<svg xmlns='http://www.w3.org/2000/svg' width='24' height='24'><circle cx='12' cy='12' r='10' fill='red'/></svg>|>
"""

if __name__ == "__main__":
  Gui(page).run(title="1157 Status with icons")

```

## Checklist

- [x] Does this solution meet the acceptance criteria of the related issue?
- [ ] Is the related issue checklist completed?
- [x] Does this PR adds unit tests for the developed code? If not, why?
- [ ] End-to-End tests have been added or updated?
- [ ] Was the documentation updated, or a dedicated issue for documentation created? (If applicable)
- [ ] Is the release notes updated? (If applicable)

### CODE REVIEW

**Review by FabienLelaquais — CHANGES_REQUESTED** [review]
Can you please:
- replace 'icons' -> 'use_icon'
- Make this property indexed:
     use_icon[warning] = True
     use_icon[e]=<inline_svg>|URL

**Review by dinhlongviolin1 — APPROVED** [review]
Looks all good to me!

File: doc/gui/examples/controls/status_mixed_icons.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    ("success", "Task Succeeded"),
+    ("info", "Process was cancelled.")
+]
+
+# Info: svg icon (pants.svg)
+# success: no icon
+# warning: default icon
+# error: inline svg icon (red disc)
+page = """
+<|{status}|status|don't use_icon|use_icon[info]=https://www.svgrepo.com/show/530594/pants.svg|use_icon[success]|use_icon[error]=<svg xmlns='http://www.w3.org/2000/svg' width='24' height='24'><circle cx='12' cy='12' r='10' fill='red'/></svg>|>
```
Comment (FabienLelaquais) [inline doc/gui/examples/controls/status_mixed_icons.py:16]: don't so that icons are initially removed everywhere hu?

**Review by FabienLelaquais — APPROVED** [review]
All examples should have the same header, potentially with a few line of specific information for that code.
See doc/gui/examples/broadcast_callback.py for example...

File: doc/gui/examples/controls/status_mixed_icons.py (hunk 1 above)
Comment (FredLL-Avaiga) [reply doc/gui/examples/controls/status_mixed_icons.py:16]: no it's the default value

Comment (FredLL-Avaiga) [reply doc/gui/examples/controls/status_mixed_icons.py:16]: in this case, it applies only to `warning`

**Pull request merged**